pilot/pkg/bootstrap/server.go

```diff
@@ -1224,7 +1226,51 @@ func (s *Server) initMeshHandlers() {
 	})
 }
 
-func (s *Server) initWorkloadTrustBundle() {
+func (s *Server) addIstioCAToTrustBundle(args *PilotArgs) {
+	// TODO: Will need to watch secrets dynamically
+	// mirrors createIstioCA
+
+	signingKeyFile := path.Join(LocalCertDir.Get(), "ca-key.pem")
+	if _, err := os.Stat(signingKeyFile); err != nil && s.kubeClient != nil {
+		// Fetch self signed certificates
+		err := retry.UntilSuccess(func() error {
```

Comment:
this is intended only for tests AFAIK

Reply:
Yes. However, unit tests fail without this. Given that this is a short term fix, I did not want to change the unit testing code to accommodate this change.

Reply:
I think it will mess with the dependencies to import the library at all - this happened in the past and we added a regression test which somehow isn't failing...